torch/_dynamo/symbolic_convert.py

```diff
@@ -2628,6 +2628,55 @@ def __init__(
         )
         linecache.lazycache(f_code.co_filename, f_globals)
 
+    def _init_torch_function_state(self):
+        # This is annoyingly complicated because of how the torch function subclass + mode C API was designed
+        # There are two exposed C knobs here as contexts: torch._C.DisableTorchFunction and torch._C.DisableTorchFunctionSubclass
+        # These are their definitions:
+        # 1) torch._C._is_torch_function_enabled indicates that neither of the above knobs have been entered
+        # (if either are entered, this will be False)
+        # 2) torch._C._is_torch_function_mode_enabled indicates that either the torch mode stack is empty OR
+        # torch._C.DisableTorchFunction has been entered
+        # To disambiguate these and keep myself sane I added a C API to check whether all torch function
+        # concepts (modes and subclasses) are enabled.
+        # This only returns true iff we have not entered torch._C.DisableTorchFunction and allows us to separate
+        # the stack length from the enablement state of torch function modes.
+        # This is important because now if a mode is pushed while dynamo is tracing, we know whether
+        # or not torch function modes are enabled and whether we should trace it.
+        self.torch_function_subclass_enabled = torch._C._is_torch_function_enabled()
```

Comment:
How hard is it to expose some more C knobs that are named the right thing?

Reply:
I exposed the extra one I needed here: https://github.com/pytorch/pytorch/pull/133136

For BC I left the existing API as is and added this new one which allows you to query all of the different  disablement states separately.